dali/operators/audio/resample.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_AUDIO_RESAMPLE_H_
+#define DALI_OPERATORS_AUDIO_RESAMPLE_H_
+
+#include <vector>
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/kernels/signal/resampling.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+namespace audio {
+
+#define AUDIO_RESAMPLE_TYPES int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t, float
+
+template <typename Backend>
+class ResampleBase : public Operator<Backend> {
+ public:
+  explicit ResampleBase(const OpSpec &spec) : Operator<Backend>(spec) {
+    DALI_ENFORCE(in_rate_.HasValue() == out_rate_.HasValue(),
+      "The parameters ``in_rate`` and ``out_rate`` must be specified together.");
+    if (in_rate_.HasValue() + scale_.HasValue() + out_length_.HasValue() > 1)
+      DALI_FAIL("The sampling rates, ``scale`` and ``out_length`` cannot be used together.");
+    if (!in_rate_.HasValue() && !scale_.HasValue() && !out_length_.HasValue())
+      DALI_FAIL("No resampling factor specified! Please supply either the scale, "
+        "the output length or the input and output sampling rates.");
+    quality_ = spec_.template GetArgument<float>("quality");
+    DALI_ENFORCE(quality_ >= 0 && quality_ <= 100, make_string("``quality`` out of range: ",
+      quality_, "\nValid range is [0..100]."));
+    if (spec_.TryGetArgument(dtype_, "dtype")) {
+      // silence useless warning -----------------------------vvvvvvvvvvvvvvvv
+      TYPE_SWITCH(dtype_, type2id, T, (AUDIO_RESAMPLE_TYPES), (T x; (void)x;),
+      (DALI_FAIL(make_string("Unsupported output type: ", dtype_,
```

Comment:
```suggestion
      TYPE_SWITCH(dtype_, type2id, T, (AUDIO_RESAMPLE_TYPES),
      (T x; (void)x;),
      (DALI_FAIL(make_string("Unsupported output type: ", dtype_,
```
I think this reads better for the `TYPE_SWITCH`.